### PR TITLE
[BEAM-11948] Drop support for Flink 1.8 and 1.9

### PR DESCRIPTION
**Pull request by iemejia**
R: @dmvk @ibzib

### CODE REVIEW

**Comment by iemejia** [comment]
Run Flink ValidatesRunner

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Streaming

File: website/www/site/content/en/documentation/runners/flink.md [hunk 1]
```diff
@@ -197,8 +197,10 @@ The optional `flink_version` option may be required as well for older versions o
 {{< paragraph class="language-portable" >}}
 Starting with Beam 2.18.0, pre-built Flink Job Service Docker images are available at Docker Hub:
 [Flink 1.8](https://hub.docker.com/r/apache/beam_flink1.8_job_server),
-[Flink 1.9](https://hub.docker.com/r/apache/beam_flink1.9_job_server).
-[Flink 1.10](https://hub.docker.com/r/apache/beam_flink1.10_job_server).
+[Flink 1.9](https://hub.docker.com/r/apache/beam_flink1.9_job_server),
```
Comment (ibzib) [inline website/www/site/content/en/documentation/runners/flink.md:200]: Looks like 1.8 and 1.9 are still listed here, not sure if that was intentional?

File: website/www/site/content/en/documentation/runners/flink.md [hunk 2]
```diff
@@ -197,8 +197,10 @@ The optional `flink_version` option may be required as well for older versions o
...
+[Flink 1.10](https://hub.docker.com/r/apache/beam_flink1.10_job_server),
+[Flink 1.10](https://hub.docker.com/r/apache/beam_flink1.11_job_server),
```
Comment (ibzib) [inline website/www/site/content/en/documentation/runners/flink.md:202]: 1.11, and 1.12 have "1.10" as their link text.

**Review by ibzib — COMMENTED** [review]
Dont' forget to add a note to CHANGES.md.

File: website/www/site/content/en/documentation/runners/flink.md (hunk 1 above)
Comment (iemejia) [reply website/www/site/content/en/documentation/runners/flink.md:200]: It was but I think maybe having there only the maintained versions makes more sense. I will fix.

File: website/www/site/content/en/documentation/runners/flink.md (hunk 2 above)
Comment (iemejia) [reply website/www/site/content/en/documentation/runners/flink.md:202]: Oups copy/paste mistake, will fix. Thanks

**Comment by iemejia** [comment]
Run Flink ValidatesRunner

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Review by ibzib — APPROVED** [review]
LGTM

**Comment by iemejia** [comment]
Rebased to catch the VR issue fix

**Comment by iemejia** [comment]
Run Flink ValidatesRunner

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Streaming

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java [hunk 3]
```diff
@@ -772,14 +771,7 @@ public void translateNode(
...
 
-        if (FlinkCapabilities.supportsOutputDuringClosing()) {
-          outputDataSet =
-              new FlatMapOperator<>(inputDataSet, typeInformation, doFnWrapper, fullName);
-        } else {
-          // This can be removed once we drop support for 1.8 and 1.9 versions.
-          outputDataSet =
-              new MapPartitionOperator<>(inputDataSet, typeInformation, doFnWrapper, fullName);
-        }
+        outputDataSet = new FlatMapOperator<>(inputDataSet, typeInformation, doFnWrapper, fullName);
```
Comment (dmvk) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java:774]: I think we can now remove MapPartition interface from https://github.com/apache/beam/blob/master/runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkDoFnFunction.java#L65

**Review by dmvk — APPROVED** [review]
LGTM 👍

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java (hunk 3 above)
Comment (iemejia) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java:774]: Oh good catch. I will fix, thx!

**Comment by iemejia** [comment]
Run Flink ValidatesRunner

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Comment by iemejia** [comment]
This one should not be merged until Beam 2.29.0 release is released

**Comment by iemejia** [comment]
Rebased to remove the warning introduced in 14214

**Comment by iemejia** [comment]
Run Flink ValidatesRunner

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Batch

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Comment by iemejia** [comment]
Failing PVR Streaming tests are unrelated to this PR for ref #14313

**Comment by iemejia** [comment]
Run Python_PVR_Flink PreCommit

**Comment by iemejia** [comment]
Run Java Flink PortableValidatesRunner Streaming

**Comment by iemejia** [comment]
I was waiting in case there was a further need to recut the 2.29.0 branch so this did not get accidentally in but it seems the 2.29.0 branch is now stable, so we can go ahead and get this one ready for 2.30.0.

I will merge it and if any issue on 2.29.0 requires a branch recut I will backport this PR as a revert. Thanks everyone !

**Pull request merged**